### PR TITLE
Combine version strings for multi-module dependencies

### LINKED ISSUE #229: [FEATURE] Don't duplicate dependency versions in build.gradle
### Is your feature request related to a problem?

Dependabot is producing separate PRs for the api and implementation for log4j and junit-jupiter.

> Should change the api version to match the engine version.  Is there a way we can just define a constant higher up in the file so we only have to update once?  Same thing happened with the two log4j updates (api and implementation).
> 
> _Originally posted by @dbwiddis in https://github.com/opensearch-project/opensearch-sdk-java/pull/228#pullrequestreview-1166232331_

### What solution would you like?

Define a version constant and use it instead of separate version numbers on the implementation line.  See [here](https://github.com/opensearch-project/opensearch-java/blob/c7b753e0daf784abbb9c519326c27b472b27f43a/java-client/build.gradle.kts#L140-L165) for example.

### PR DESCRIPTION
**Pull request by dbwiddis**
Signed-off-by: Daniel Widdis <widdis@gmail.com>

### Description
Reduces the noise of dependabot updates by only having a single version to update.

### Issues Resolved
Fixes #229 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by owaiskazi19** [comment]
Should we keep a variable for `implementation "org.apache.lucene:lucene-core:9.4.0-snapshot-ddf0d0a"`?

File: build.gradle [hunk 1]
```diff
@@ -81,8 +83,8 @@ dependencies {
     implementation("com.fasterxml.jackson.core:jackson-databind:${jacksonDatabindVersion}")
     implementation("com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:${jacksonDatabindVersion}")
     implementation("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:${jacksonDatabindVersion}")
-    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.8.2'
-    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.8.2'
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:${junit5Version}'
```
Comment (owaiskazi19) [inline build.gradle:86]: Let's keep this in the same format as below
```
testImplementation("org.opensearch.test:framework:${opensearchVersion}")

File: build.gradle [hunk 2]
```diff
@@ -62,13 +62,15 @@ configurations {
...
+    def log4jVersion = "2.19.0"
     def jacksonDatabindVersion = "2.12.6.1"
     def opensearchVersion = "3.0.0-SNAPSHOT"
+    def junit5Version = "5.9.1"
 
     implementation("org.opensearch:opensearch:${opensearchVersion}")
     implementation("org.opensearch.plugin:transport-netty4-client:${opensearchVersion}")
-    implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: '2.17.1'
-    implementation group: 'org.apache.logging.log4j', name: 'log4j-core', version: '2.19.0'
+    implementation group: 'org.apache.logging.log4j', name: 'log4j-api', version: '${log4jVersion}'
```
Comment (owaiskazi19) [inline build.gradle:72]: Same format as above
```
implementation("org.apache.logging.log4j:log4j-api:${log4jVersion}")

**Comment by dbwiddis** [comment]
> Should we keep a variable for `implementation "org.apache.lucene:lucene-core:9.4.0-snapshot-ddf0d0a"`?

My usual preference is to have _all_ versions as separate properties.

My fix commit crossed paths with your comments -- do you want me to change _all_ dependencies in the file to the same format?

**Comment by owaiskazi19** [comment]
> My fix commit crossed paths with your comments -- do you want me to change _all_ dependencies in the file to the same format?
Looks like you already did. Looks clean now!

**Review by owaiskazi19 — APPROVED** [review]
Thanks for formatting this!

**Review by joshpalis — APPROVED** [review]
Thanks Dan!

**Pull request merged**